crates/pessimistic-proof/src/global_state.rs

```diff
@@ -0,0 +1,67 @@
+use std::collections::BTreeMap;
+
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    certificate::Certificate,
+    local_balance_tree::BalanceTree,
+    local_exit_tree::{hasher::Keccak256Hasher, LocalExitTree},
+    proof::{BalanceRoot, ExitRoot},
+    BridgeExit, NetworkId, ProofError,
+};
+
+/// Local state of one network.
+/// The AggLayer tracks the [`LocalNetworkState`] for all networks.
+/// Eventually, this state will be entirely tracked by the networks themselves.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct LocalNetworkState {
+    /// Commitment to the [`BridgeExit`].
+    pub exit_tree: LocalExitTree<Keccak256Hasher>,
+    /// Commitment to the balance for each token.
+    pub balance_tree: BalanceTree,
+    /// Commitment to the imported [`BridgeExit`].
+    #[allow(dead_code)]
+    pub nullifier_tree: BTreeMap<NetworkId, Vec<BridgeExit>>,
```

Comment:
Can we have a dedicated struct or type for the `NullifierTree` ?

Reply:
I am removing this placeholder, I'll add all the nullifier tree related things in another PR